### PR TITLE
SonarCloud Analysis + Improved Workflows -- Frontend + Backend

### PR DESCRIPTION
**Pull request by iustin05**

### CODE REVIEW

File: .github/workflows/backend_dotnet.yml [hunk 1]
```diff
@@ -27,4 +28,95 @@ jobs:
...
+    runs-on: windows-latest
+    needs: sonarcloud
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '7.x'
+          include-prerelease: true
```
Comment (IonutAdriann) [inline .github/workflows/backend_dotnet.yml:85]: We could use a better practice like Github CodeQL or Snyk

File: .github/workflows/frontend_dotnet.yml [hunk 2]
```diff
@@ -27,3 +28,95 @@ jobs:
...
+        with:
+          name: .net-app
+
+      - name: Deploy to Azure Web App
+        id: deploy-to-webapp
+        uses: azure/webapps-deploy@v2
+        with:
+          app-name: 'sep3group1'
+          slot-name: 'Production'
+          publish-profile: ${{ secrets.AZUREAPPSERVICE_PUBLISHPROFILE_FD6537FBB8374569ABB1C47E688EA238 }}
```
Comment (IonutAdriann) [inline .github/workflows/frontend_dotnet.yml:121]: Same here

File: .github/workflows/backend_dotnet.yml (hunk 1 above)
Comment (IonutAdriann) [reply .github/workflows/backend_dotnet.yml:85]: Github CodeQL might be free with the Pro subscription

**Comment by IonutAdriann** [comment]
Also there are some conflicts

File: .github/workflows/backend_dotnet.yml (hunk 1 above)
Comment (iustin05) [reply .github/workflows/backend_dotnet.yml]: Created another issue regarding changing from SonarCloud to CodeQL

File: .github/workflows/frontend_dotnet.yml (hunk 2 above)
Comment (iustin05) [reply .github/workflows/frontend_dotnet.yml]: Created another issue regarding changing from SonarCloud to CodeQL

**Review by IonutAdriann — APPROVED** [review]

**Pull request merged**